snuba/web/delete_query.py

```diff
@@ -27,23 +30,68 @@ def construct_condition(body: Dict[str, Any]) -> Expression:
     return combine_and_conditions(and_conditions)
 
 
+def _get_rows_to_delete(
+    storage: WritableTableStorage, select_query_to_count_rows: Query
+) -> int:
+    formatted_select_query_to_count_rows = format_query(select_query_to_count_rows)
+    select_query_results = (
+        storage.get_cluster().get_reader().execute(formatted_select_query_to_count_rows)
+    )
+    return typing.cast(int, select_query_results["data"][0]["count"])
+
+
+def _enforce_max_rows(
+    storage: WritableTableStorage,
+    from_clause: Table,
+    condition: Expression,
+    on_cluster: Literal | None,
+) -> None:
+    select_query_to_count_rows = Query(
+        selected_columns=[
+            SelectedExpression("count", FunctionCall("count", "count", ())),
+        ],
+        from_clause=from_clause,
+        condition=condition,
+        on_cluster=on_cluster,
+        is_delete=False,
+    )
+    rows_to_delete = _get_rows_to_delete(
+        storage=storage, select_query_to_count_rows=select_query_to_count_rows
+    )
+    max_rows_allowed = storage.get_deletion_settings().max_rows_to_delete
+    if rows_to_delete > max_rows_allowed:
+        raise MaxRowsEnforcerException(
+            f"Query wants to delete {rows_to_delete} rows, but the maximum allowed is {max_rows_allowed}"
+        )
+
+
 def delete_query(
     storage: WritableTableStorage, table: str, body: Dict[str, Any]
 ) -> Result:
     cluster_name = storage.get_cluster().get_clickhouse_cluster_name()
     on_cluster = literal(cluster_name) if cluster_name else None
+    from_clause = Table(
+        table,
+        ColumnSet([]),
+        storage_key=storage.get_storage_key(),
+        # TODO: add allocation policies
+        allocation_policies=[],
```

Comment:
FYI: even if you add allocation policies here they still will not run because of how you're querying them

https://github.com/getsentry/snuba/pull/6142/files#diff-c3016e51e59d45df20ed861d6db6b02d99925816fd18b5cebe581a6c44e684dbR38

no action needed, just FYI